src/monodroid/jni/monodroid-glue.c

```diff
@@ -100,6 +101,21 @@ static int wait_for_gdb;
 static volatile int monodroid_gdb_wait = TRUE;
 static int android_api_level = 0;
 
+// Values correspond to the CPU_KIND_* macros
+static char* android_abi_names[] = {
```

Comment:
I think it might be handy to explicitly provide the array size:

```c
static const char* android_abi_names[CPU_KIND_X86_64+1] = {
```

That way if we add a `CPU_KIND_FOO` value and an element to the array, we'll get a warning unless we update the array size:

```c
/* some time later... */
#define CPU_KIND_X86_64  ((unsigned short)6)

/* ... */
static const char *android_abi_names[CPU_KIND_X86_64+1] = {
  /* ... */
  "foo", /* added */
};
/* results in a compiler warning:
warning: excess elements in array initializer
  "foo",
  ^---
 */
```

What this *won't* do is ensure that indexes match as we expect.  `clang` supports this syntax; not sure about `gcc`/C99:

```c
static const char * android_abi_names[CPU_KIND_X86_64+1] = {
  "*unknown*",
  [CPU_KIND_ARM]      = "armeabi-v7a",
  [CPU_KIND_ARM64]    = "arm64-v8a",
  [CPU_KIND_MIPS]     = "mips",
  [CPU_KIND_X86]      = "x86",
  [CPU_KIND_X86_64]   = "x86_64",
};
```

Reply:
Good idea!